src/deploy/osp_deployer/settings_sanity.py

```diff
@@ -489,3 +489,54 @@ def verify_iha_dependency_on_fencing(self):
     def validate_profile(self):
         self.profile = Profile()
         self.profile.validate_configuration()
+
+    def validate_nic_configs(self):
+
+        # Get the user supplied NIC settings from the .ini
+        ini_nics_settings = self.settings.get_nics_settings()
+
+        missing_settings = []
+        nic_env_file_missing = False
+        if 'nic_env_file' not in ini_nics_settings:
+            nic_env_file_missing = True
+            missing_settings.append('nic_env_file')
+
+        if 'sah_bond_opts' not in ini_nics_settings:
+            missing_settings.append('sah_bond_opts')
+
+        extra_settings = set()
+        missing_variable_settings = set()
+        if not nic_env_file_missing:
+            # Get the NIC settings from the NIC environment file
+            with open(self.settings.nic_env_file_path, 'r') as yaml_stream:
```

Comment:
Maybe check if the file exists before trying to open it; not an issue for XSP and CSP profiles , but could potentially be for a custom profile/custom nic file

Reply:
Done.